src/restore.c

```diff
@@ -153,13 +153,58 @@ restore_main(int32_t argc, char **argv)
 	}
 
 	uint32_t line_no;
+	as_vector directories;
+	as_vector_init(&directories, sizeof(char*), 1);
+
+	// restoring from multiple directories
+	if (conf.directory_list != NULL) {
+
+		uint32_t dir_count = 0;
+
+		char *dir_clone = safe_strdup(conf.directory_list);
+		split_string(dir_clone, ',', false, &directories);
+		dir_count = directories.size;
+
+		for (uint32_t i = 0; i < dir_count; i++) {
+			char *dir = as_vector_get_ptr(&directories, i);
+
+			if (conf.parent_directory) {
+				size_t path_size = strlen(conf.parent_directory) + strlen(dir) + 1;
+				char *tmp_dir = dir;
+				dir = cf_malloc(path_size);
+				snprintf(dir, path_size, "%s%s", conf.parent_directory, tmp_dir);
```

Comment:
If the paths in --directory-list start with '/' it will work fine, otherwise the path parsing fails. Concatenating the strings gives users the most freedom.

Reply:
Yeah, but most applications don't care whether you add or remove the trailing `/`. I think it would be a nicer user experience if they can just provide them without having to know whether to add the trailing `/`. All I am saying is, I would be annoyed if the problem was because I needed to add an extra `/`.  I would think there is a simple `path_join(...)` but probably not because this is c :). As long as the error message is clear I am good with it

Reply:
Since the option is called "parent-directory" I think this is reasonable to do. It was easy enough so added it.